### PR TITLE
fix: add runtime hints for spanner 

### PR DESCRIPTION
**Pull request by mpeddada1**
Changes included in this PR are:

- RuntimeHints for SpannerTemplateIntegrationTests to resolve 
```
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'tradeRepositoryTransactionalService': Unsatisfied dependency expressed through field 'tradeRepository': Error creating bean with name 'tradeRepository': No suitable constructor found on class com.google.cloud.spring.data.spanner.repository.support.SimpleSpannerRepository to match the given arguments: com.google.cloud.spring.data.spanner.core.SpannerTemplate, java.lang.Class. Make sure you implement a constructor taking these
	at org.springframework.beans.factory.aot.AutowiredFieldValueResolver.resolveValue(AutowiredFieldValueResolver.java:195)
	at org.springframework.beans.factory.aot.AutowiredFieldValueResolver.resolveObject(AutowiredFieldValueResolver.java:154)
	at org.springframework.beans.factory.aot.AutowiredFieldValueResolver.resolve(AutowiredFieldValueResolver.java:143)

```
- RuntimeHints for CommitTimestampsIntegrationTests to resolve:
```
JUnit Jupiter:CommitTimestampIntegrationTests:testCommitTimestamp()
    MethodSource [className = 'com.google.cloud.spring.data.spanner.core.convert.it.CommitTimestampIntegrationTests', methodName = 'testCommitTimestamp', methodParameterTypes = '']
    => com.google.cloud.spanner.SpannerException: INVALID_ARGUMENT: io.grpc.StatusRuntimeException: INVALID_ARGUMENT: Invalid StreamingRead request.
field_violations {
  field: "columns"
  description: "Column list must not be empty for streaming Read requests."
}

       com.google.cloud.spanner.SpannerExceptionFactory.newSpannerExceptionPreformatted(SpannerExceptionFactory.java:291)
       com.google.cloud.spanner.SpannerExceptionFactory.newSpannerExceptionPreformatted(SpannerExceptionFactory.java:297)
       com.google.cloud.spanner.SpannerExceptionFactory.newSpannerException(SpannerExceptionFactory.java:170)
       com.google.cloud.spanner.SpannerExceptionFactory.newSpannerException(SpannerExceptionFactory.java:110)
       com.google.cloud.spanner.AbstractResultSet$GrpcStreamIterator.computeNext(AbstractResultSet.java:1028)
       [...]
```
- Avoiding SpyBean usage in SpannerRepositoryIntegrationTests to address AOT processing failure. SpyBean and MockBean are [not supported in Spring Boot native image compilation ](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-with-GraalVM#testing). 
```
19:26:51.663 [main] WARN  o.s.t.c.aot.TestContextAotGenerator - Failed to generate AOT artifacts for test classes [com.google.cloud.spring.data.spanner.repository.it.SpannerRepositoryIntegrationTests]. Enable DEBUG logging to view the stack trace. org.springframework.test.context.aot.TestContextAotException: Failed to process test class [com.google.cloud.spring.data.spanner.repository.it.SpannerRepositoryIntegrationTests] for AOT

```

### CODE REVIEW

File: spring-cloud-gcp-data-spanner/src/test/java/com/google/cloud/spring/data/spanner/repository/it/SpannerRepositoryIntegrationTests.java [hunk 1]
```diff
@@ -488,9 +480,6 @@ void queryMethodsTest_EagerFetch() {
         .hasValueSatisfying(t -> assertThat(t.getSymbol()).isEqualTo(aTrade.getSymbol()))
         .hasValueSatisfying(
             t -> assertThat(t.getSubTrades()).hasSize(aTrade.getSubTrades().size()));
-    Mockito.verify(spannerTemplate, Mockito.times(1)).executeQuery(any(Statement.class), any());
-    Mockito.verify(spannerTemplate, Mockito.times(1))
-        .query(eq(Trade.class), any(Statement.class), any(SpannerQueryOptions.class));
```
Comment (mpeddada1) [inline spring-cloud-gcp-data-spanner/src/test/java/com/google/cloud/spring/data/spanner/repository/it/SpannerRepositoryIntegrationTests.java:493]: These lines verify whether the `executeQuery` and `query` are called with the given parameter types which should be implicitly checked in the lines above.  From the style guide, "Verification should only be done if calling that exact method is crucial to the correct functioning of the code" which makes me think that we are over-asserting.  Looking for a second opinion on this?

File: spring-cloud-gcp-data-spanner/src/main/java/com/google/cloud/spring/data/spanner/aot/SpannerRuntimeHints.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  @Override
+  public void registerHints(RuntimeHints hints, ClassLoader classLoader) {
+    hints
+        .reflection()
+        .registerTypes(
+            Arrays.asList(
+                TypeReference.of(
+                    com.google.cloud.spring.data.spanner.repository.support.SimpleSpannerRepository
+                        .class),
+                TypeReference.of(java.lang.String.class)),
```
Comment (burkedavison) [inline spring-cloud-gcp-data-spanner/src/main/java/com/google/cloud/spring/data/spanner/aot/SpannerRuntimeHints.java:36]: Needing to include `String` here seems odd. Do we know why this is needed?

**Review by burkedavison — APPROVED** [review]

File: spring-cloud-gcp-data-spanner/src/main/java/com/google/cloud/spring/data/spanner/aot/SpannerRuntimeHints.java (hunk 2 above)
Comment (mpeddada1) [reply spring-cloud-gcp-data-spanner/src/main/java/com/google/cloud/spring/data/spanner/aot/SpannerRuntimeHints.java]: Included a comment with some context but looking at the stacktrace, this type is referenced from our code`SpannerPersistentEntityImpl.tableName()` which then leads to a reflective call down the line within the Spring code:

 https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/c92260f1b583575436a61ec8f6fb2252737cd43f/spring-cloud-gcp-data-spanner/src/main/java/com/google/cloud/spring/data/spanner/core/mapping/SpannerPersistentEntityImpl.java#L351-L360:
```
com.oracle.svm.core.jdk.UnsupportedFeatureError: Runtime reflection is not supported for public java.lang.String java.lang.String.concat(java.lang.String)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.util.VMError.unsupportedFeature(VMError.java:89)
	at java.base@17.0.7/java.lang.reflect.Method.acquireMethodAccessor(Method.java:76)
	at java.base@17.0.7/java.lang.reflect.Method.invoke(Method.java:566)
	at org.springframework.expression.spel.support.ReflectiveMethodExecutor.execute(ReflectiveMethodExecutor.java:139)
	at org.springframework.expression.spel.ast.MethodReference.getValueInternal(MethodReference.java:112)
	at org.springframework.expression.spel.ast.MethodReference$MethodValueRef.getValue(MethodReference.java:380)
	at org.springframework.expression.spel.ast.CompoundExpression.getValueInternal(CompoundExpression.java:96)
	at org.springframework.expression.spel.ast.SpelNodeImpl.getTypedValue(SpelNodeImpl.java:119)
	at org.springframework.expression.spel.standard.SpelExpression.getValue(SpelExpression.java:309)
	at com.google.cloud.spring.data.spanner.core.mapping.SpannerPersistentEntityImpl.tableName(SpannerPersistentEntityImpl.java:361)
	at com.google.cloud.spring.data.spanner.core.admin.SpannerSchemaUtils.getDropTableDdlString(SpannerSchemaUtils.java:85)
	at com.google.cloud.spring.data.spanner.core.admin.SpannerSchemaUtils.lambda$getDropTableDdlStringsForInterleavedHierarchy$1(SpannerSchemaUtils.java:280)
	at com.google.cloud.spring.data.spanner.core.admin.SpannerSchemaUtils.getDdlStringForInterleavedHierarchy(SpannerSchemaUtils.java:297)
	at com.google.cloud.spring.data.spanner.core.admin.SpannerSchemaUtils.getDropTableDdlStringsForInterleavedHierarchy(SpannerSchemaUtils.java:275)
	at com.google.cloud.spring.data.spanner.core.admin.SpannerSchemaUtils.getDropTableDdlStringsForInterleavedHierarchy(SpannerSchemaUtils.java:160)

```

File: spring-cloud-gcp-data-spanner/src/test/java/com/google/cloud/spring/data/spanner/repository/it/SpannerRepositoryIntegrationTests.java (hunk 1 above)
Comment (mpeddada1) [reply spring-cloud-gcp-data-spanner/src/test/java/com/google/cloud/spring/data/spanner/repository/it/SpannerRepositoryIntegrationTests.java:493]: Updating this comment for posterity: As discussed offline, verified that this check is redundant as the # invocations for `executeQuery` are covered through unit tests under `SqlSpannerQueryTests`.

**Pull request merged**